drivers/serial/uart_cc23x0.c

```diff
@@ -0,0 +1,405 @@
+		struct uart_cc23x0_data *data = dev->data;                                         \
+		int ret;                                                                           \
+                                                                                                   \
+		CLKCTLEnable(CLKCTL_BASE, CLKCTL_UART0);                                           \
+                                                                                                   \
+		ret = pinctrl_apply_state(data->pcfg, PINCTRL_STATE_DEFAULT);                      \
+		if (ret < 0) {                                                                     \
+			return ret;                                                                \
+		}                                                                                  \
+                                                                                                   \
+		/* Configure and enable UART */                                                    \
+		ret = uart_cc23x0_configure(dev, &data->uart_config);                              \
+                                                                                                   \
+		/* Enable interrupts */                                                            \
+		UART_CC23X0_IRQ_CFG(n);                                                            \
+                                                                                                   \
+		return ret;                                                                        \
+	}
+
+#define UART_CC23X0_INIT(n)                                                                        \
+	PINCTRL_DT_INST_DEFINE(n);                                                                 \
+	UART_CC23X0_INIT_FUNC(n);                                                                  \
+                                                                                                   \
+	static const struct uart_cc23x0_config uart_cc23x0_config_##n = {                          \
+		.reg = DT_INST_REG_ADDR(n),                                                        \
+		.sys_clk_freq = DT_INST_PROP_BY_PHANDLE(n, clocks, clock_frequency)};              \
+                                                                                                   \
+	static struct uart_cc23x0_data uart_cc23x0_data_##n = {                                    \
+		.uart_config =                                                                     \
+			{                                                                          \
+				.baudrate = DT_INST_PROP(n, current_speed),                        \
+				.parity = UART_CFG_PARITY_NONE,                                    \
+				.stop_bits = UART_CFG_STOP_BITS_1,                                 \
+				.data_bits = UART_CFG_DATA_BITS_8,                                 \
+				.flow_ctrl = UART_CFG_FLOW_CTRL_NONE,                              \
```

Comment:
these are defined in `uart-controller.yaml`, why not use them instead of hardcoding them?